File: packages/shared-types/src/dtos/tasks/task.dto.ts
import { TaskStatusType } from '../../enums/task-status.enum';
import { TaskPriorityType } from '../../enums/task-priority.enum';
import { EntityTimestamps } from '../../types/common.types';

/**
 * Task DTO
 */
export interface TaskDto extends EntityTimestamps {
  id: string;
  title: string;
  description: string | null;
  taskType: 'regular' | 'meeting' | 'milestone';
  status: TaskStatusType;
  priority: TaskPriorityType | null;
  columnId: string;
  boardId: string;
  projectId: string;
  goalId: string | null;
  position: number;
  dueDate: string | null;  // YYYY-MM-DD
  estimatedMinutes: number | null;
  actualMinutes: number | null;
  filePath: string | null;
  completedAt: string | null;
}

/**
 * Task with context (board, project names)
 */
export interface TaskDetailDto extends TaskDto {
  boardName: string;
  projectName: string;
  columnName: string;
}

/**
 * Create task request
 */
export interface TaskCreateRequestDto {
  title: string;
  description?: string;
  taskType?: 'regular' | 'meeting' | 'milestone';
  priority?: TaskPriorityType;
  columnId: string;
  goalId?: string;
  position?: number;
  dueDate?: string;
  estimatedMinutes?: number;
}

/**
 * Update task request
 */
export interface TaskUpdateRequestDto {
  title?: string;
  description?: string;
  taskType?: 'regular' | 'meeting' | 'milestone';
  status?: TaskStatusType;
  priority?: TaskPriorityType;
  columnId?: string;
  goalId?: string;
  position?: number;
  dueDate?: string;
  estimatedMinutes?: number;
  actualMinutes?: number;
  completedAt?: string;
}

/**
 * Move task request
 */
export interface TaskMoveRequestDto {
  columnId: string;
  position: number;
}

File: apps/tui/internal/application/dto/task_dto.go
package dto

type TaskDto struct {
	ID               string     `json:"id"`
	Slug             string     `json:"slug"`
	Title            string     `json:"title"`
	Description      *string    `json:"description"`
	TaskType         string     `json:"taskType"`
	Status           string     `json:"status"`
	Priority         *string    `json:"priority"`
	ColumnID         string     `json:"columnId"`
	Column           *ColumnDto `json:"column,omitempty"`
	BoardID          string     `json:"boardId"`
	ProjectID        string     `json:"projectId"`
	GoalID           *string    `json:"goalId"`
	Position         int        `json:"position"`
	DueDate          *string    `json:"dueDate"`
	EstimatedMinutes *int       `json:"estimatedMinutes"`
	ActualMinutes    *int       `json:"actualMinutes"`
	FilePath         *string    `json:"filePath"`
	CompletedAt      *string    `json:"completedAt"`
	ParentID         *string    `json:"parentId"`
	CreatedAt        string     `json:"createdAt"`
	UpdatedAt        string     `json:"updatedAt"`
}

type TaskCreateRequest struct {
	Title       string  `json:"title"`
	Description *string `json:"description,omitempty"`
	TaskType    *string `json:"taskType,omitempty"`
	Priority    *string `json:"priority,omitempty"`
	ColumnID    string  `json:"columnId"`
	DueDate     *string `json:"dueDate,omitempty"`
	ParentID    *string `json:"parentId,omitempty"`
}

type TaskQuickCreateRequest struct {
	Title    string `json:"title"`
	ColumnID string `json:"columnId"`
}

type TaskUpdateRequest struct {
	Title            *string `json:"title,omitempty"`
	Description      *string `json:"description,omitempty"`
	Priority         *string `json:"priority,omitempty"`
	Status           *string `json:"status,omitempty"`
	DueDate          *string `json:"dueDate,omitempty"`
	EstimatedMinutes *int    `json:"estimatedMinutes,omitempty"`
}

type TaskMoveRequest struct {
	TargetColumnID string `json:"targetColumnId"`
}
